### PR TITLE
[bug](function) fix is_null/is_not_null check is_const has error

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
In this PR #20158 
In order to solve core dump, so convert const column to full column.
But the root reason of this case is is_const() function have check error about is_null/is_not_null function.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 1]
```diff
@@ -1575,6 +1575,16 @@ && collectChildReturnTypes()[0].isDecimalV3()) {
             }
         }
 
+        // if the input arg is not nullable, so the function of is_null/is_not_null
+        // result is very certain and there is no need to look at data in column.
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:1579]: use ExprRewriteRule rewrite `is_null_pred(not_nullable_column)` to `false`

File: be/src/vec/exprs/vcompound_pred.h [hunk 2]
```diff
@@ -65,8 +65,7 @@ class VCompoundPred : public VectorizedFnCall {
         int lhs_id = -1;
         int rhs_id = -1;
         RETURN_IF_ERROR(_children[0]->execute(context, block, &lhs_id));
-        ColumnPtr lhs_column =
-                block->get_by_position(lhs_id).column->convert_to_full_column_if_const();
+        ColumnPtr lhs_column = block->get_by_position(lhs_id).column;
```
Comment (BiteTheDDDDt) [inline be/src/vec/exprs/vcompound_pred.h:68]: Do not remove defensive code

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run p0 feut

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run p0

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**